src/components/Spawn/MountVolumeSelect.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from "react";
+import { useQuery } from "@apollo/client";
+import { Select } from "antd";
+import { ModalContent } from "components/Spawn";
+import { InputLabel } from "components/styles";
+import { pollInterval } from "constants/index";
+import { useBannerDispatchContext } from "context/banners";
+import { MyHostsQuery, MyHostsQueryVariables } from "gql/generated/types";
+import { GET_MY_HOSTS } from "gql/queries";
+import { useNetworkStatus } from "hooks/useNetworkStatus";
+
+const { Option } = Select;
+interface HostOption {
+  key: string;
+  displayName: string;
+  label?: string;
+}
+
+interface Props {
+  targetAvailabilityZone: string;
+  selectedHostId: string;
+  onChange: (hostId: string) => void;
+  label?: string;
+  autofill?: boolean;
+}
+
+export const MountVolumeSelect = ({
+  targetAvailabilityZone,
+  selectedHostId,
+  onChange,
+  label,
+  autofill,
+}: Props) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [hostOptions, setHostOptions] = useState<HostOption[]>([]); // dropdown option
+  const { data, startPolling, stopPolling } = useQuery<
+    MyHostsQuery,
+    MyHostsQueryVariables
+  >(GET_MY_HOSTS, {
+    pollInterval,
+    onError: (e) => {
+      dispatchBanner.errorBanner(
+        `There was an error loading hosts: ${e.message}`
+      );
+    },
+  });
+  useNetworkStatus(startPolling, stopPolling);
+
+  // set host dropdown options
+  useEffect(() => {
+    if (data?.myHosts) {
+      const opts = data.myHosts
+        // Filter hosts that do not have the same availability zone as the volume.
+        .filter(
+          ({ availabilityZone }) => availabilityZone === targetAvailabilityZone
+        )
+        // Map host to a displayName and ID for the dropdown <Option />
+        .map((host) => ({
+          key: host.id,
```

Comment:
Why rename id to key?

Reply:
There was no specific reason for this. I will change key to id to make the code more readable.